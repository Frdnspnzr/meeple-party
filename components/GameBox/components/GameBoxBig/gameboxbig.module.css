.card {
  width: 600px;
  height: 252px;
  display: grid;
  background-color: var(--bs-gray-100);
  grid-template-columns: 250px 60px 290px;
  grid-template-rows: 4rem auto;
  grid-template-areas:
    "image info title"
    "image info status";
}

.image {
  width: 250px;
  height: 250px;
  grid-area: image;
  object-fit: cover;
}

.name {
  grid-area: title;
  padding: 0.5rem;
}

.name h2 {
  font-size: 1.25rem;
  line-height: 1.5rem;
  max-height: 3rem;
  overflow: hidden;
  font-weight: bold;
  margin: 0;
}

.info {
  grid-area: info;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-evenly;
  border-right: 1px solid var(--bs-border-color);
}

.metricDivider {
  width: 80%;
  margin: 0;
}

.metricTitle {
  font-size: 1.25rem;
  text-align: center;
  font-weight: bold;
}

.metricLabel {
  font-size: 0.6rem;
  text-align: center;
  color: var(--bs-light-text);
}

.statusBox {
  grid-area: status;
  background-color: lightgray;
}

.statusList {
  height: 100%;
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
}

.status {
  display: grid;
  grid-template-rows: 2fr 32px 3fr;
  grid-template-areas:
    "icon"
    "text"
    "friends";
}

.statusButton {
  font-size: 1.75rem;
  place-self: center stretch;
}

.statusText {
  font-size: 0.7rem;
  text-align: center;
  font-weight: lighter;
}
