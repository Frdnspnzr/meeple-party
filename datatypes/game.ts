import * as client from "@prisma/client";

/**
 * @deprecated This type should be replaced by the Data Layer V2 version
 */
export type Game = Omit<client.Game, "updatedAt">;

/**
 * @deprecated This type should be replaced by the Data Layer V2 version
 */
export interface BggGame extends Game {
  description: string;
  designers: string[];
  artists: string[];
  age: number;
}
