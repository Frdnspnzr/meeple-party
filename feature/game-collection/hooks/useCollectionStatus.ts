import { showError } from "@/lib/utility/error";
import { GameCollection } from "@prisma/client";
import {
  UseMutationResult,
  useMutation,
  useQuery,
  useQueryClient,
} from "@tanstack/react-query";
import axios from "axios";
import { Result } from "../../../lib/types/apiHooks";

const twoWeeksInMilliSeconds = 1000 * 60 * 60 * 24 * 14;

type CollectionStatusUpdate = Partial<
  Pick<GameCollection, "own" | "wantToPlay" | "wishlist">
>;

function getCollectionStatus(gameId: number) {
  console.log("Loading game collection", gameId);
  return axios
    .get<GameCollection>(`/api/v2/game/${gameId}/collection`)
    .then((response) => response.data);
}

interface UseCollectionStatus extends Result<GameCollection> {
  updateMutation: UseMutationResult<CollectionStatusUpdate, Error, any, any>;
}

export default function useCollectionStatus(
  gameId: number
): UseCollectionStatus {
  const queryClient = useQueryClient();
  const { isLoading, isError, data } = useQuery({
    queryKey: getCollectionStatusQueryKey(gameId),
    queryFn: () => getCollectionStatus(gameId),
    refetchOnWindowFocus: false,
    staleTime: twoWeeksInMilliSeconds,
  });

  const updateMutation = useMutation({
    mutationKey: getCollectionStatusQueryKey(gameId),
    mutationFn: (data: CollectionStatusUpdate) => {
      return axios
        .patch<CollectionStatusUpdate>(
          `/api/v2/game/${gameId}/collection`,
          data
        )
        .then((response) => response.data);
    },
    onMutate: async (newData) => {
      const queryKey = getCollectionStatusQueryKey(gameId);
      await queryClient.cancelQueries({ queryKey });
      const previousData = queryClient.getQueryData(queryKey);
      queryClient.setQueryData(queryKey, (oldData: GameCollection) => ({
        ...oldData,
        ...newData,
      }));
      return { previousData };
    },
    onSuccess: (newData) => {
      queryClient.setQueryData(getCollectionStatusQueryKey(gameId), {
        ...data,
        ...newData,
      });
    },
    onError: (error: Error, newData, context) => {
      console.error(error);
      showError({ message: "Could not update collection status." });
      queryClient.setQueryData(
        getCollectionStatusQueryKey(gameId),
        context?.previousData
      );
    },
  });

  return {
    isLoading,
    isError,
    data,
    invalidate: () => {
      queryClient.invalidateQueries({
        queryKey: getCollectionStatusQueryKey(gameId),
      });
    },
    updateMutation,
  };
}

export function getCollectionStatusQueryKey(gameId: number): any[] {
  return ["collectionstatus", gameId];
}
