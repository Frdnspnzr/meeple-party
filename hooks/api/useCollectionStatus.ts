import { GameCollection } from "@prisma/client";
import { useMutation, useQuery, useQueryClient } from "@tanstack/react-query";
import axios from "axios";
import { MutableResult } from "./types";

const twoWeeksInMilliSeconds = 1000 * 60 * 60 * 24 * 14;

type CollectionStatusUpdate = Partial<
  Omit<GameCollection, "userId" | "gameId" | "updatedAt">
>;

function getCollectionStatus(gameId: number) {
  console.log("Loading game collection", gameId);
  return axios
    .get<GameCollection>(`/api/v2/game/${gameId}/collection`)
    .then((response) => response.data);
}

export default function useCollectionStatus(
  gameId: number
): MutableResult<GameCollection> {
  const queryClient = useQueryClient();
  const {
    isLoading: queryLoading,
    isError: queryError,
    data: queryData,
  } = useQuery({
    queryKey: getCollectionStatusQueryKey(gameId),
    queryFn: () => getCollectionStatus(gameId),
    refetchOnWindowFocus: false,
    staleTime: twoWeeksInMilliSeconds,
  });

  const {
    isLoading: mutationLoading,
    isError: mutationError,
    mutate: mutationFunction,
  } = useMutation({
    mutationKey: getCollectionStatusQueryKey(gameId),
    mutationFn: (data: CollectionStatusUpdate) => {
      return axios
        .patch<CollectionStatusUpdate>(
          `/api/v2/game/${gameId}/collection`,
          data
        )
        .then((response) => response.data);
    },
    onSuccess: (data) => {
      queryClient.setQueryData(getCollectionStatusQueryKey(gameId), {
        ...queryData,
        ...data,
      });
    },
  });

  return {
    isLoading: queryLoading || mutationLoading,
    isError: queryError || mutationError,
    data: queryData,
    invalidate: () => {
      queryClient.invalidateQueries({
        queryKey: getCollectionStatusQueryKey(gameId),
      });
    },
    mutate: (data, configuration) => mutationFunction(data),
  };
}

export function getCollectionStatusQueryKey(gameId: number): any[] {
  return ["collectionstatus", gameId];
}
