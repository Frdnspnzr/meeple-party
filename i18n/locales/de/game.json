{
    "Page": {
        "Header": {
            "Design": "Design",
            "Art": "Illustration",
            "Players": "Spielende",
            "Minutes": "Minuten",
            "Weight": "Komplexität"
        },
        "Collections": {
            "YourCollection": "Deine Sammlung",
            "FriendsOwn": "Im Besitz",
            "FriendsWantToPlay": "Wollen spielen",
            "FriendsWishlist": "Wünschen sich"
        }
    }
}