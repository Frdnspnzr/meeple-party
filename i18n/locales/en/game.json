{
    "Attributes": {
        "Players": "Players",
        "PlayingTime": "Playing Time",
        "Weight": "Weight"
    },
    "Page": {
        "Header": {
            "Design": "Design",
            "Art": "Art",
            "Players": "players",
            "Minutes": "minutes",
            "Weight": "weight"
        },
        "Collections": {
            "YourCollection": "Your collection",
            "FriendsOwn": "Owned by",
            "FriendsWantToPlay": "Want to play",
            "FriendsWishlist": "Wishlist"
        },
        "More": "â€¦ and {{count}} more"
    }
}