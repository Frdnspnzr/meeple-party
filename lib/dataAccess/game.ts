import { GameGetResult } from "@/app/api/v2/game/[gameId]/route";
import { Game as ServerGame } from "@prisma/client";
import axios from "axios";
import { Game } from "../datatypes/client/game";

export function getGame(gameId: number): Promise<Game> {
  return axios
    .get<GameGetResult>(`/api/v2/game/${gameId}`)
    .then((response) => response.data.game)
    .then(convertGame);
}

export function convertGame(serverGame: ServerGame): Game {
  return {
    id: serverGame.id,
    maxPlayers: serverGame.maxPlayers,
    minPlayers: serverGame.minPlayers,
    name: serverGame.name,
    playingTime: serverGame.playingTime,
    weight: serverGame.weight,
    year: serverGame.year,
    BGGRank: serverGame.BGGRank || undefined,
    BGGRating: serverGame.BGGRating || undefined,
    image: serverGame.image || undefined,
    thumbnail: serverGame.thumbnail || undefined,
  };
}
