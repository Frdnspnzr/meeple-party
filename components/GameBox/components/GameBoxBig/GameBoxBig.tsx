"use client";

import AvatarStack from "@/components/AvatarStack/AvatarStack";
import PersonList from "@/components/PersonList/PersonList";
import StatusButton from "@/components/StatusButton/StatusButton";
import { useModal } from "@/context/modalContext";
import { StatusByUser } from "@/datatypes/collection";
import { Game } from "@/datatypes/game";
import { UserProfile } from "@/datatypes/userProfile";
import useCollectionStatus from "@/hooks/api/useCollectionStatus";
import { useTranslation } from "@/i18n/client";
import classNames from "classnames";
import Image from "next/image";
import { useCallback, useMemo } from "react";
import { GameBoxProps } from "../../GameBox";
import styles from "./gameboxbig.module.css";

export default function GameBoxBig({
  game,
  friendCollection,
}: Readonly<GameBoxProps>) {
  return (
    <div className={classNames("card", styles.card)}>
      {game.image ? (
        <Image
          src={game.image}
          alt={game.name}
          width={250}
          height={250}
          unoptimized
          className={classNames(styles.image, "rounded-start")}
        />
      ) : (
        <div className={classNames(styles.image, "rounded-start")} />
      )}
      <MetricList game={game} />
      <h2 className={styles.name}>{game.name}</h2>
      <StatusList game={game} friendCollection={friendCollection} />
    </div>
  );
}

function MetricList({ game }: Readonly<{ game: Game }>) {
  const { t } = useTranslation("game");
  return (
    <div className={styles.info}>
      <Metric
        text={round(game.weight).toString()}
        label={t("Attributes.Weight")}
      />
      <hr className={styles.metricDivider} />
      <Metric
        text={game.playingTime.toString()}
        label={t("Attributes.PlayingTime")}
      />
      <hr className={styles.metricDivider} />
      <Metric
        text={
          game.minPlayers === game.maxPlayers
            ? game.minPlayers.toString()
            : `${game.minPlayers}-${game.maxPlayers}`
        }
        label={t("Attributes.Players")}
      />
    </div>
  );
}

function Metric({ text, label }: Readonly<{ text: string; label: string }>) {
  return (
    <div className={styles.metric}>
      <div className={styles.metricTitle}>{text}</div>
      <div className={styles.metricLabel}>{label}</div>
    </div>
  );
}

function StatusList({
  game,
  friendCollection,
}: Readonly<{
  game: Game;
  friendCollection?: StatusByUser;
}>) {
  return (
    <>
      <Status
        game={game}
        status="own"
        friends={friendCollection ? friendCollection.own : []}
      />
      <Status
        game={game}
        status="wanttoplay"
        friends={friendCollection ? friendCollection.wantToPlay : []}
      />
      <Status
        game={game}
        status="wishlist"
        friends={friendCollection ? friendCollection.wishlist : []}
      />
    </>
  );
}

function Status({
  game,
  friends,
  status,
}: Readonly<{
  game: Game;
  friends: UserProfile[];
  status: "own" | "wishlist" | "wanttoplay";
}>) {
  const { t } = useTranslation("default");
  const { t: ct } = useTranslation("collection");
  const { data } = useCollectionStatus(game.id);
  const { open: openModal } = useModal();
  const state = useMemo(() => {
    if (!data) {
      return false;
    } else if (status === "own") {
      return data.own;
    } else if (status === "wanttoplay") {
      return data.wantToPlay;
    } else if (status === "wishlist") {
      return data.wishlist;
    }
  }, [data, status]);

  const translationBaseKey = useMemo(() => {
    if (status === "own") {
      return "Own";
    } else if (status === "wanttoplay") {
      return "WantToPlay";
    } else if (status === "wishlist") {
      return "Wishlist";
    }
  }, [status]);

  const showFriends = useCallback(() => {
    openModal({
      title: ct(`FriendCollections.${translationBaseKey}`, {
        game: game.name,
      }),
      content: <PersonList persons={friends} />,
    });
  }, [friends, game.name, ct, openModal, translationBaseKey]);

  return (
    <>
      <div
        className={classNames({
          [styles.myStatusOwn]: status === "own",
          [styles.myStatusWantToPlay]: status === "wanttoplay",
          [styles.myStatusWishlist]: status === "wishlist",
        })}
      >
        <StatusButton
          status={status}
          gameId={game.id}
          className={styles.statusButton}
        />
        <div className={styles.statusText}>
          {state
            ? t(`States.${translationBaseKey}`)
            : t(`States.Not${translationBaseKey}`)}
        </div>
      </div>
      {friends.length > 0 && (
        <div
          className={classNames({
            [styles.friendStatusOwn]: status === "own",
            [styles.friendStatusWantToPlay]: status === "wanttoplay",
            [styles.friendStatusWishlist]: status === "wishlist",
          })}
          onClick={showFriends}
          onKeyDown={(e) => {
            if (e.key === "Enter") {
              showFriends();
            }
          }}
          tabIndex={0}
          data-testid={`friends-${translationBaseKey}`}
        >
          <AvatarStack
            max={4}
            distance="sm"
            avatars={friends.map((f) => ({
              id: f.id,
              name: f.name || "",
              image: f.image,
            }))}
          />
          <div className={styles.statusText}>
            {t(`FriendStates.${translationBaseKey}`, { count: friends.length })}
          </div>
        </div>
      )}
    </>
  );
}

function round(x: number): number {
  return Math.round(x * 10) / 10;
}
