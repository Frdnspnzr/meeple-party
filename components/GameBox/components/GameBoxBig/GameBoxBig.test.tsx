import { Game } from "@/datatypes/game";
import { render } from "@testing-library/react";
import useCollectionStatus from "../../../../hooks/api/useCollectionStatus";
import GameBoxBig from "./GameBoxBig";

jest.mock("@/i18n/client");
jest.mock("@/hooks/api/useCollectionStatus");

const game: Game = {
  id: 123,
  name: "Test game",
  BGGRank: 23,
  BGGRating: 8.4,
  minPlayers: 1,
  maxPlayers: 5,
  playingTime: 120,
  weight: 3.8,
  year: 2018,
  image: "testimage",
  thumbnail: "testthumbnail",
};

describe("GameBoxBig", () => {
  it("Matches snapshot", async () => {
    jest.mocked(useCollectionStatus).mockReturnValue({
      invalidate: () => {},
      isError: false,
      isLoading: false,
      mutate: () => {},
      data: {
        gameId: 123,
        own: true,
        userId: "abc123",
        wantToPlay: true,
        wishlist: true,
        updatedAt: null,
      },
    });
    const { container: rendered } = render(
      <GameBoxBig game={game} showFriendCollection />
    );
    expect(rendered).toMatchSnapshot("");
  });
  it("Matches snapshot without friend collection", async () => {
    jest.mocked(useCollectionStatus).mockReturnValue({
      invalidate: () => {},
      isError: false,
      isLoading: false,
      mutate: () => {},
      data: {
        gameId: 123,
        userId: "abc123",
        own: true,
        wantToPlay: true,
        wishlist: true,
        updatedAt: null,
      },
    });
    const { container: rendered } = render(
      <GameBoxBig game={game} showFriendCollection={false} />
    );
    expect(rendered).toMatchSnapshot("default game");
  });
  it("Matches snapshot when not in collection", async () => {
    jest.mocked(useCollectionStatus).mockReturnValue({
      invalidate: () => {},
      isError: false,
      isLoading: false,
      mutate: () => {},
      data: {
        gameId: 123,
        userId: "abc123",
        own: false,
        wantToPlay: false,
        wishlist: false,
        updatedAt: null,
      },
    });
    const { container: rendered } = render(
      <GameBoxBig game={game} showFriendCollection />
    );
    expect(rendered).toMatchSnapshot("default game");
  });
  it("Matches snapshot for precise weight", async () => {
    jest.mocked(useCollectionStatus).mockReturnValue({
      invalidate: () => {},
      isError: false,
      isLoading: false,
      mutate: () => {},
      data: {
        gameId: 123,
        userId: "abc123",
        own: false,
        wantToPlay: false,
        wishlist: false,
        updatedAt: null,
      },
    });
    const { container: rendered } = render(
      <GameBoxBig game={{ ...game, weight: Math.PI }} showFriendCollection />
    );
    expect(rendered).toMatchSnapshot("default game");
  });
});
