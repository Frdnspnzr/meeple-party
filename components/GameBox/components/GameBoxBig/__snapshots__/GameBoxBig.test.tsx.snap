// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameBoxBig Matches snapshot 1`] = `
<div>
  <div
    class="card card"
  >
    <img
      alt="Test game"
      class="image rounded-start"
      data-nimg="1"
      decoding="async"
      height="250"
      loading="lazy"
      src="testimage"
      style="color: transparent;"
      width="250"
    />
    <div
      class="info"
    >
      <div
        class="metric"
      >
        <div
          class="metricTitle"
        >
          3.8
        </div>
        <div
          class="metricLabel"
        >
          Attributes.Weight
        </div>
      </div>
      <hr
        class="metricDivider"
      />
      <div
        class="metric"
      >
        <div
          class="metricTitle"
        >
          120
        </div>
        <div
          class="metricLabel"
        >
          Attributes.PlayingTime
        </div>
      </div>
      <hr
        class="metricDivider"
      />
      <div
        class="metric"
      >
        <div
          class="metricTitle"
        >
          1-5
        </div>
        <div
          class="metricLabel"
        >
          Attributes.Players
        </div>
      </div>
    </div>
    <h2
      class="name"
    >
      Test game
    </h2>
    <div
      class="myStatusOwn"
    >
      <div
        class="button statusButton"
        role="button"
        tabindex="0"
      >
        <i
          class="bi bi-box-seam-fill"
        />
      </div>
      <div
        class="statusText"
      >
        States.Own
      </div>
    </div>
    <div
      class="friendStatusOwn"
      tabindex="0"
    >
      <div
        class="container sm"
      >
        <div
          class="additional"
        >
          +
          6
        </div>
        <div
          class="dummy avatar"
          style="background-color: rgb(10, 200, 128);"
        >
          T
        </div>
        <div
          class="dummy avatar"
          style="background-color: rgb(64, 178, 239);"
        >
          B
        </div>
        <div
          class="dummy avatar"
          style="background-color: rgb(184, 28, 197);"
        >
          M
        </div>
        <div
          class="dummy avatar"
          style="background-color: rgb(67, 24, 96);"
        >
          M
        </div>
      </div>
      <div
        class="statusText"
      >
        FriendStates.Own
      </div>
    </div>
    <div
      class="myStatusWantToPlay"
    >
      <div
        class="button statusButton"
        role="button"
        tabindex="0"
      >
        <i
          class="bi bi-dice-3-fill"
        />
      </div>
      <div
        class="statusText"
      >
        States.WantToPlay
      </div>
    </div>
    <div
      class="friendStatusWantToPlay"
      tabindex="0"
    >
      <div
        class="container sm"
      >
        <div
          class="additional"
        >
          +
          6
        </div>
        <div
          class="dummy avatar"
          style="background-color: rgb(149, 93, 222);"
        >
          C
        </div>
        <div
          class="dummy avatar"
          style="background-color: rgb(220, 14, 100);"
        >
          J
        </div>
        <div
          class="dummy avatar"
          style="background-color: rgb(108, 59, 31);"
        >
          C
        </div>
        <div
          class="dummy avatar"
          style="background-color: rgb(75, 225, 169);"
        >
          B
        </div>
      </div>
      <div
        class="statusText"
      >
        FriendStates.WantToPlay
      </div>
    </div>
    <div
      class="myStatusWishlist"
    >
      <div
        class="button statusButton"
        role="button"
        tabindex="0"
      >
        <i
          class="bi bi-gift-fill"
        />
      </div>
      <div
        class="statusText"
      >
        States.Wishlist
      </div>
    </div>
    <div
      class="friendStatusWishlist"
      tabindex="0"
    >
      <div
        class="container sm"
      >
        <div
          class="additional"
        >
          +
          6
        </div>
        <div
          class="dummy avatar"
          style="background-color: rgb(184, 233, 64);"
        >
          H
        </div>
        <div
          class="dummy avatar"
          style="background-color: rgb(143, 96, 67);"
        >
          S
        </div>
        <div
          class="dummy avatar"
          style="background-color: rgb(121, 207, 74);"
        >
          6
        </div>
        <div
          class="dummy avatar"
          style="background-color: rgb(173, 84, 72);"
        >
          I
        </div>
      </div>
      <div
        class="statusText"
      >
        FriendStates.Wishlist
      </div>
    </div>
  </div>
</div>
`;

exports[`GameBoxBig Matches snapshot for precise weight: default game 1`] = `
<div>
  <div
    class="card card"
  >
    <img
      alt="Test game"
      class="image rounded-start"
      data-nimg="1"
      decoding="async"
      height="250"
      loading="lazy"
      src="testimage"
      style="color: transparent;"
      width="250"
    />
    <div
      class="info"
    >
      <div
        class="metric"
      >
        <div
          class="metricTitle"
        >
          3.1
        </div>
        <div
          class="metricLabel"
        >
          Attributes.Weight
        </div>
      </div>
      <hr
        class="metricDivider"
      />
      <div
        class="metric"
      >
        <div
          class="metricTitle"
        >
          120
        </div>
        <div
          class="metricLabel"
        >
          Attributes.PlayingTime
        </div>
      </div>
      <hr
        class="metricDivider"
      />
      <div
        class="metric"
      >
        <div
          class="metricTitle"
        >
          1-5
        </div>
        <div
          class="metricLabel"
        >
          Attributes.Players
        </div>
      </div>
    </div>
    <h2
      class="name"
    >
      Test game
    </h2>
    <div
      class="myStatusOwn"
    >
      <div
        class="button statusButton"
        role="button"
        tabindex="0"
      >
        <i
          class="bi bi-box-seam-fill"
        />
      </div>
      <div
        class="statusText"
      >
        States.NotOwn
      </div>
    </div>
    <div
      class="myStatusWantToPlay"
    >
      <div
        class="button statusButton"
        role="button"
        tabindex="0"
      >
        <i
          class="bi bi-dice-3-fill"
        />
      </div>
      <div
        class="statusText"
      >
        States.NotWantToPlay
      </div>
    </div>
    <div
      class="myStatusWishlist"
    >
      <div
        class="button statusButton"
        role="button"
        tabindex="0"
      >
        <i
          class="bi bi-gift-fill"
        />
      </div>
      <div
        class="statusText"
      >
        States.NotWishlist
      </div>
    </div>
  </div>
</div>
`;

exports[`GameBoxBig Matches snapshot when not in collection: default game 1`] = `
<div>
  <div
    class="card card"
  >
    <img
      alt="Test game"
      class="image rounded-start"
      data-nimg="1"
      decoding="async"
      height="250"
      loading="lazy"
      src="testimage"
      style="color: transparent;"
      width="250"
    />
    <div
      class="info"
    >
      <div
        class="metric"
      >
        <div
          class="metricTitle"
        >
          3.8
        </div>
        <div
          class="metricLabel"
        >
          Attributes.Weight
        </div>
      </div>
      <hr
        class="metricDivider"
      />
      <div
        class="metric"
      >
        <div
          class="metricTitle"
        >
          120
        </div>
        <div
          class="metricLabel"
        >
          Attributes.PlayingTime
        </div>
      </div>
      <hr
        class="metricDivider"
      />
      <div
        class="metric"
      >
        <div
          class="metricTitle"
        >
          1-5
        </div>
        <div
          class="metricLabel"
        >
          Attributes.Players
        </div>
      </div>
    </div>
    <h2
      class="name"
    >
      Test game
    </h2>
    <div
      class="myStatusOwn"
    >
      <div
        class="button statusButton"
        role="button"
        tabindex="0"
      >
        <i
          class="bi bi-box-seam-fill"
        />
      </div>
      <div
        class="statusText"
      >
        States.NotOwn
      </div>
    </div>
    <div
      class="myStatusWantToPlay"
    >
      <div
        class="button statusButton"
        role="button"
        tabindex="0"
      >
        <i
          class="bi bi-dice-3-fill"
        />
      </div>
      <div
        class="statusText"
      >
        States.NotWantToPlay
      </div>
    </div>
    <div
      class="myStatusWishlist"
    >
      <div
        class="button statusButton"
        role="button"
        tabindex="0"
      >
        <i
          class="bi bi-gift-fill"
        />
      </div>
      <div
        class="statusText"
      >
        States.NotWishlist
      </div>
    </div>
  </div>
</div>
`;

exports[`GameBoxBig Matches snapshot without friend collection: default game 1`] = `
<div>
  <div
    class="card card"
  >
    <img
      alt="Test game"
      class="image rounded-start"
      data-nimg="1"
      decoding="async"
      height="250"
      loading="lazy"
      src="testimage"
      style="color: transparent;"
      width="250"
    />
    <div
      class="info"
    >
      <div
        class="metric"
      >
        <div
          class="metricTitle"
        >
          3.8
        </div>
        <div
          class="metricLabel"
        >
          Attributes.Weight
        </div>
      </div>
      <hr
        class="metricDivider"
      />
      <div
        class="metric"
      >
        <div
          class="metricTitle"
        >
          120
        </div>
        <div
          class="metricLabel"
        >
          Attributes.PlayingTime
        </div>
      </div>
      <hr
        class="metricDivider"
      />
      <div
        class="metric"
      >
        <div
          class="metricTitle"
        >
          1-5
        </div>
        <div
          class="metricLabel"
        >
          Attributes.Players
        </div>
      </div>
    </div>
    <h2
      class="name"
    >
      Test game
    </h2>
    <div
      class="myStatusOwn"
    >
      <div
        class="button statusButton"
        role="button"
        tabindex="0"
      >
        <i
          class="bi bi-box-seam-fill"
        />
      </div>
      <div
        class="statusText"
      >
        States.Own
      </div>
    </div>
    <div
      class="friendStatusOwn"
      tabindex="0"
    >
      <div
        class="container sm"
      >
        <div
          class="additional"
        >
          +
          6
        </div>
        <div
          class="dummy avatar"
          style="background-color: rgb(124, 52, 187);"
        >
          H
        </div>
        <div
          class="dummy avatar"
          style="background-color: rgb(59, 232, 13);"
        >
          Q
        </div>
        <div
          class="dummy avatar"
          style="background-color: rgb(95, 115, 111);"
        >
          D
        </div>
        <div
          class="dummy avatar"
          style="background-color: rgb(189, 180, 34);"
        >
          G
        </div>
      </div>
      <div
        class="statusText"
      >
        FriendStates.Own
      </div>
    </div>
    <div
      class="myStatusWantToPlay"
    >
      <div
        class="button statusButton"
        role="button"
        tabindex="0"
      >
        <i
          class="bi bi-dice-3-fill"
        />
      </div>
      <div
        class="statusText"
      >
        States.WantToPlay
      </div>
    </div>
    <div
      class="friendStatusWantToPlay"
      tabindex="0"
    >
      <div
        class="container sm"
      >
        <div
          class="additional"
        >
          +
          6
        </div>
        <div
          class="dummy avatar"
          style="background-color: rgb(254, 0, 208);"
        >
          Z
        </div>
        <div
          class="dummy avatar"
          style="background-color: rgb(219, 65, 233);"
        >
          Y
        </div>
        <div
          class="dummy avatar"
          style="background-color: rgb(173, 186, 134);"
        >
          F
        </div>
        <div
          class="dummy avatar"
          style="background-color: rgb(204, 20, 200);"
        >
          N
        </div>
      </div>
      <div
        class="statusText"
      >
        FriendStates.WantToPlay
      </div>
    </div>
    <div
      class="myStatusWishlist"
    >
      <div
        class="button statusButton"
        role="button"
        tabindex="0"
      >
        <i
          class="bi bi-gift-fill"
        />
      </div>
      <div
        class="statusText"
      >
        States.Wishlist
      </div>
    </div>
    <div
      class="friendStatusWishlist"
      tabindex="0"
    >
      <div
        class="container sm"
      >
        <div
          class="additional"
        >
          +
          6
        </div>
        <div
          class="dummy avatar"
          style="background-color: rgb(207, 70, 180);"
        >
          S
        </div>
        <div
          class="dummy avatar"
          style="background-color: rgb(37, 138, 153);"
        >
          D
        </div>
        <div
          class="dummy avatar"
          style="background-color: rgb(211, 69, 27);"
        >
          T
        </div>
        <div
          class="dummy avatar"
          style="background-color: rgb(107, 110, 164);"
        >
          P
        </div>
      </div>
      <div
        class="statusText"
      >
        FriendStates.Wishlist
      </div>
    </div>
  </div>
</div>
`;
